packages/core/src/common/preferences/preference-schema.ts

```diff
@@ -76,6 +76,7 @@ export interface PreferenceItem {
     additionalProperties?: object | boolean;
     [name: string]: any;
     overridable?: boolean;
+    hidden?: boolean;
```

Comment:
@westbury the property is used to `hide` elements in the view but it still allows users to add the preference in their settings (with or without the use of auto-completion). Is there a reason you want to hide the preference rather than make it non-editable?

Reply:
> Is there a reason you want to hide the preference rather than make it non-editable?

Consider a custom IDE that is targeted to a very specific set of users using a specific tool chain for a specific purpose.  There will be preferences that must be set to a particular value and setting the preference to any other value will always break the user.  For example, a couple of days ago, after I had put in this PR, we had a bug report from a user. They had set on the 'Support Multi Root Workspace' preference.  Our IDE works to a specific directory layout and does not support multi-root workspaces.  So, in this situation, should we show the preference but make it non-editable, or not show it?  I would suggest that not showing it is a better user experience.  Mentioning 'multi-root workspaces' in a custom IDE that has no concept of multi-root workspaces will only cause confusion.  If users know what multi-root workspaces are then they will think they must be supported and send in support requests complaining that they can't get them to work.  A large part of providing a clean IDE is that we do not expose stuff that does not apply.

Reply:
Probably should be removed from auto-complete in settings too.

Reply:
@westbury I'm not arguing the use-case, I believe it is something I brought up in the past: https://spectrum.chat/theia/general/is-there-a-need-to-set-specific-preferences-as-non-overridable~2cf3bb68-59b2-4f4b-8070-b2839c35ba4d.

The issue with the approach is while the preference is `hidden`, the preference is still editable which can still break applications (for instance if they do not expect to support multi-root workspaces). We must also confirm that such preferences are not updated through the `package.json` (which is used to override the default value).

Reply:
I don't see how we can stop the user from editing the settings file.  If the requirement is that we do not want to give the user any way of changing the preference (short of hacking javascript) then we would have to ignore any value in the settings file, always using the default value.  That would provide a consistent approach but it does remove the ability to change the preference value programatically.

The current implementation leaves the hidden property in the schema.  The property would be better hidden if it were not in the schema, as it would not then appear in auto-complete.  A disadvantage of removing hidden properties from the schema is that if the preference has been set programatically then the user will see it with squiggly underlining.

If we decide it is okay that hidden properties cannot be changed programatically (we have no use case) then it should be removed from the schema.